dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/imports/preheat/supplier/strategy/RelationshipStrategy.java

```diff
@@ -27,30 +27,42 @@
  */
 package org.hisp.dhis.tracker.imports.preheat.supplier.strategy;
 
+import com.google.common.collect.Lists;
+import java.util.ArrayList;
 import java.util.List;
-import java.util.stream.Collectors;
-import javax.annotation.Nonnull;
-import lombok.RequiredArgsConstructor;
+import javax.persistence.EntityManager;
 import org.hisp.dhis.common.CodeGenerator;
-import org.hisp.dhis.relationship.RelationshipStore;
-import org.hisp.dhis.tracker.imports.domain.Relationship;
+import org.hisp.dhis.hibernate.HibernateGenericStore;
+import org.hisp.dhis.relationship.Relationship;
 import org.hisp.dhis.tracker.imports.preheat.TrackerPreheat;
 import org.hisp.dhis.tracker.imports.preheat.mappers.RelationshipMapper;
 import org.hisp.dhis.tracker.imports.preheat.supplier.DetachUtils;
+import org.springframework.context.ApplicationEventPublisher;
+import org.springframework.jdbc.core.JdbcTemplate;
 import org.springframework.stereotype.Component;
 
 /**
  * @author Luciano Fiandesio
  */
-@RequiredArgsConstructor
 @Component
-@StrategyFor(value = Relationship.class, mapper = RelationshipMapper.class)
-public class RelationshipStrategy implements ClassBasedSupplierStrategy {
-  @Nonnull private final RelationshipStore relationshipStore;
+@StrategyFor(
+    value = org.hisp.dhis.tracker.imports.domain.Relationship.class,
+    mapper = RelationshipMapper.class)
+public class RelationshipStrategy extends HibernateGenericStore<Relationship>
+    implements ClassBasedSupplierStrategy {
+
+  public RelationshipStrategy(
+      EntityManager entityManager,
+      JdbcTemplate jdbcTemplate,
+      ApplicationEventPublisher publisher,
+      Class<Relationship> clazz,
```

Comment:
Is this argument always the same for this class? if so can we inline that and remove one constructor param